### ISSUE TITLE
Exception "index is out over bounds" with custom grid

### ISSUE THREAD

**Issue by ghelle-fhh** [t-rex-tileserver/t-rex#124]
Hi,
I'm having some difficulties with a custom grid for my tiles.

My data is in a PostGIS DB and in EPSG:25832 and I'm trying to use a grid definition with the following parameters:

[grid.user]
width = 256
height = 256
extent = { minx = 454591, miny = 5809000, maxx = 738000, maxy = 6075769 }
srid = 25832
units = "m"
resolutions = [66.14614761460263,
26.458319045841044,
15.874991427504629,
10.583327618336419,
5.2916638091682096,
2.6458319045841048,
1.3229159522920524,
0.6614579761460262,
0.2645831904584105]
origin = "TopLeft"

But I receive an error when I run the server saying "the index is out over bounds, the len is 9 but the index is 9"

![grafik](https://user-images.githubusercontent.com/51916535/59663707-e3e12c00-91af-11e9-80dd-c9241c58b9d5.png)

Any ideas?
Thanks!

**Comment by schmity** [comment]
Hi, 
same issue here, any hints very much appreciated.

Best, Sebastian

**Comment by ghelle-fhh** [comment]
So I've tried digging deeper and tried the example locally:
https://github.com/t-rex-tileserver/t-rex/blob/master/examples/usergrid.toml

The only difference is I used the geopackage instead of connecting to a database.

the map appears, but it does not seem to work as expected:
![custom_grid](https://user-images.githubusercontent.com/51916535/61453831-761b5200-a95f-11e9-9cc9-c4cb8d6ff225.gif)
.

**Comment by pka** [comment]
The built-in viewer does only support the Mercator grid. If you want to display other grids you have to setup OpenLayers similar to `examples/usergrid.html`

Re-reading your original question the error says `the index is out over bounds, the len is 9 but the index is 910`, which is even weirder. Could you run t_rex with `--loglevel debug` to see more details?

**Comment by ghelle-fhh** [comment]
Ahhh, of course. I'll try that out.

Running with log-level doesn't provide much more info unfortunately:
![image](https://user-images.githubusercontent.com/51916535/61457217-1d03ec00-a968-11e9-80a7-87ae61b3373d.png)

**Comment by pka** [comment]
What I see is that requests are made for level 9 and probably 10.
You have to limit the zoom level of your tileset according to the tile grid resolutions. Example:
```
[[tileset]]
#...
maxzoom = 9
```

**Comment by ghelle-fhh** [comment]
Hi there,

thank you for the tip, that solved the index issue!

I adopted your example for my projection and it worked like a charm. Although the links to the stylesheet and script for openlayers appear to be broken:
https://openlayers.org/en/v3.18.2/css/ol.css
https://openlayers.org/en/v3.18.2/build/ol-debug.js

So the example didn't work out of the box. I updated the links to Openlayers 5.3.0 and added the following line to adapt to slightly different OL 5 syntax:

`ol.proj.proj4.register(proj4);`

And it all worked like a charm.

Thanks for the help and the great tool!
